### ISSUE TITLE
can't connect sony A7R3 (camera is already in use)

### ISSUE THREAD

**Issue by pdeman** [gphoto/gphoto2#268]
Hi,
I try to use a sony A7R3 on ubuntu 16.04 (on nvidia jetson tx) and I can't. it says the device is already in use.
but if I do:
ps -aux | grep gphoto 
I have nothing. 
(as long time ago I did this:
sudo chmod -x /usr/lib/gvfs/gvfs-gphoto2-volume-monitor and sudo chmod -x /usr/lib/gvfs/gvfsd-gphoto2 )

if I do gphoto2 --summary --debug I have the file attached.
[bug.txt](https://github.com/gphoto/gphoto2/files/3902068/bug.txt)

i seemed their is an auto mount of the sony as mass storage so I did this:
systemctl stop udisks2.service.
and when I plug the sony it's not auto mounted now.

but it still does the same error: could not claim the device.

if I do:
ps -aux | grep usb-storage 
I have this:
root 6542 0.0 0.0 0 0 ? S 15:08 0:00 [usb-storage]
nvidia 6785 0.0 0.0 5548 571 pts/2 S+ 15:17 0:00 grep --color=auto usb-storage.

any idea how to solve this ?

**Comment by msmeissn** [comment]
Caan you check what USB mode the Camera  is in ? perhaps it really is the Mass Storage mode?
For remote control it needs to be in "PC Control" or similar named mode.

**Comment by pdeman** [comment]
How can i check that ? It's in the menu of the camera ?

**Comment by pdeman** [comment]
ok I finally found in the sony menu the choice between : 
mass storage
mtp
and remote pc
I tried with remote pc then.
but what bother me is that I can't choose in the menu:
PC remote settings -> still img.save dest: camera only which is the option that is interesting me.

anyway
I can't trigger an image.
if I do : 
--trigger-capture it doesn't crash but I don't see the image on the camera, neither on the pc, here is the debug file:
[testTriggerSony.txt](https://github.com/gphoto/gphoto2/files/3910762/testTriggerSony.txt)

if I do : 
--capture-and-download the command never finishes:
here is the debug file:
[testTriggerDownloadSony.txt](https://github.com/gphoto/gphoto2/files/3910767/testTriggerDownloadSony.txt)

I don't get why.

**Comment by pdeman** [comment]
same when using libusb1 instead of 0:
[testTriggerDownloadSony2.txt](https://github.com/gphoto/gphoto2/files/3910947/testTriggerDownloadSony2.txt)
[testTriggerSony2.txt](https://github.com/gphoto/gphoto2/files/3910948/testTriggerSony2.txt)

**Comment by pdeman** [comment]
here is the result of lsusb corresponding to the camera in case it helps:

Bus 001 Device 010: ID 054c:0c33 Sony Corp.
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.10
  bDeviceClass            0 (Defined at Interface level)
  bDeviceSubClass         0
  bDeviceProtocol         0
  bMaxPacketSize0        64
  idVendor           0x054c Sony Corp.
  idProduct          0x0c33
  bcdDevice            2.00
  iManufacturer           1 Sony
  iProduct                2 ILCE-7RM3
  iSerial                 3 CA792039E251
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength           39
    bNumInterfaces          1
    bConfigurationValue     1
    iConfiguration          0
    bmAttributes         0x80
      (Bus Powered)
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           3
      bInterfaceClass         6 Imaging
      bInterfaceSubClass      1 Still Image Capture
      bInterfaceProtocol      1 Picture Transfer Protocol (PIMA 15470)
      iInterface              0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x02  EP 2 OUT
        bmAttributes            2
          Transfer Type            Bulk
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0200  1x 512 bytes
        bInterval               0
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0020  1x 32 bytes
        bInterval               7
Binary Object Store Descriptor:
  bLength                 5
  bDescriptorType        15
  wTotalLength           22
  bNumDeviceCaps          2
  USB 2.0 Extension Device Capability:
    bLength                 7
    bDescriptorType        16
    bDevCapabilityType      2
    bmAttributes   0x00000002
      Link Power Management (LPM) Supported
  SuperSpeed USB Device Capability:
    bLength                10
    bDescriptorType        16
    bDevCapabilityType      3
    bmAttributes         0x00
    wSpeedsSupported   0x000e
      Device can operate at Full Speed (12Mbps)
      Device can operate at High Speed (480Mbps)
      Device can operate at SuperSpeed (5Gbps)
    bFunctionalitySupport   1
      Lowest fully-functional device speed is Full Speed (12Mbps)
    bU1DevExitLat          10 micro seconds
    bU2DevExitLat        2047 micro seconds
Device Status:     0x0000
  (Bus Powered)

**Comment by msmeissn** [comment]
Foremost, the camera only cpatures to its RAM, so the image needs to be downloaded immediately.

so the --trigger-capture itself will not download the image here (it needs additiona --wait-event-and-download=5s  or similar time)

The --capture-image-and-download however should do this. AS the logfile is incomplete, does it hang?

I see that it still uses libusb0 in the second trace, perhaps the usb0 driver still is there, can you remove it via:
rm /usr/local/lib/libgphoto2_port/0.12.0/usb*

**Comment by pdeman** [comment]
I deleted everything in libgphoto2_port and compile again.
now I don't have the usb0 anymore.
it still doesn't work but it "finishes" now.
here is the debug file:
https://drive.google.com/open?id=1hmwxR16N68d_RhBl3cw3rIwvkbmbKynR

**Comment by pdeman** [comment]
> AS the logfile is incomplete, does it hang?
it's not incomplete, it "stops" there and never finish.
I have to do Ctrl+C to abord.

> I see that it still uses libusb0 in the second trace
where do you see that ?

**Comment by pdeman** [comment]
oups I closed by error. it's not solved

**Comment by msmeissn** [comment]
Debug wise it looks to do polling now, but the image does not appear in RAM. (the d215 lines)

Did you hear the shutter click?

**Comment by pdeman** [comment]
no, but even when we use it (not remotely I mean) we don't here anything. it must be an electronic shutter.

**Comment by pdeman** [comment]
no shutter noise.
here are the log files for trigger and wait commands as well
[testWait6.txt](https://github.com/gphoto/gphoto2/files/3911774/testWait6.txt)
[testTriggerSony4.txt](https://github.com/gphoto/gphoto2/files/3911775/testTriggerSony4.txt)

**Comment by msmeissn** [comment]
what architecture is the controller device?
 I found a small bug that misdetects focusmodem, fixed in git.

if the camera in manual focus right now, can you try other focus mnodes like autofocus?

**Comment by pdeman** [comment]
controller device is nvidia jetson tx1 (arm)
the camera is in autofocus.

I'll pull soon and try again.

**Comment by pdeman** [comment]
not  better.
here is the log file for capture and download:
https://drive.google.com/open?id=1ZrtnUL6-s4Figz7YNgKiRxw_oZIRql-u

**Comment by pdeman** [comment]
here is the log file in manual focus instead of autofocus.
[testTriggerDownloadSonyGitmodifManualFocus.txt](https://github.com/gphoto/gphoto2/files/3912045/testTriggerDownloadSonyGitmodifManualFocus.txt)

**Comment by msmeissn** [comment]
the last one had usb comm troubles a bit, can you run it again (the manual focus one)?

**Comment by msmeissn** [comment]
https://drive.google.com/open?id=1ZrtnUL6-s4Figz7YNgKiRxw_oZIRql-u   looked like manu7al focus though, not auto

**Comment by msmeissn** [comment]
I tried another adjustment, as the focus wait was not triggering for you

**Comment by pdeman** [comment]
@msmeissn it was set as auto. That is strange.
I'll try again wednesday
